### PR TITLE
Create PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by spfink**
## Description
Adds a pull request template to the repo

## Motivation and Context
This change helps contributors make changes to our code base by providing a template that includes all of the details that the repo owners want to see when contributing.

## Testing
Tested in staging repository.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [X] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## License

- [X] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+
+## Checklist
+<!--- Go over all the following points, and put an `x` in all the boxes that apply -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+- [ ] My code follows the code style of this project
+- [ ] My change requires a change to the documentation
+- [ ] I have updated the documentation accordingly
```
Comment (millems) [inline .github/PULL_REQUEST_TEMPLATE.md]: Does this mean javadoc or developer guide?

Comment (spfink) [reply .github/PULL_REQUEST_TEMPLATE.md]: Should always mean javadoc.

Not many (if any) changes post GA should require updates to the dev guide. Changes that do will likely be taken on by our team separately.

Comment (shorea) [reply .github/PULL_REQUEST_TEMPLATE.md]: I wouldn't expect a customer to update the developer guide too so +1 for clarifying this as javadoc.

Comment (millems) [reply .github/PULL_REQUEST_TEMPLATE.md]: :+1:

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Screenshots (if appropriate)
+
+## Types of changes
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+
+## Checklist
```
Comment (dagnir) [inline .github/PULL_REQUEST_TEMPLATE.md]: There's some stuff we call out under "Pull Request Readiness" [`CONTRIBUTING.md`](./CONTRIBUTING.md#pull-request-readiness) that we should include here as well, like adding a changelog entry if necessary

Comment (spfink) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added

**Pull request merged**